### DISCUSSION TITLE
Q&A

### DISCUSSION THREAD

**Discussion by DNNDHH** [DNNDHH/F-D-L#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Zhen-Bo** [comment]
想好奇問下為何一些操作都要用讀/寫檔案來實現?
難道用傳值的方式他不香嗎🤔
如果說是為了要 debug 用的話我覺得進 debug 模式設個斷點可能會更好
真看不明白也能在斷點的情況下把變數手動轉存成 json
我把自己 refactor 的板本轉公開了,可以互相交流參考下 :+1:
[我 refactor 的版本 (In Progress)](https://github.com/Zhen-Bo/FGO_Automatic/tree/workflow) 
目前還有一堆 internal function 卡在一個檔案, 之後應該會抽模組, 不過現在有點懶了

EX: 
https://github.com/DNNDHH/F-D-L/blob/d47939202402696a1ecb63d1cea2c51bca4c0ebc/user.py#L147-L148
https://github.com/DNNDHH/F-D-L/blob/d47939202402696a1ecb63d1cea2c51bca4c0ebc/user.py#L363-L364
https://github.com/DNNDHH/F-D-L/blob/d47939202402696a1ecb63d1cea2c51bca4c0ebc/user.py#L371-L372
https://github.com/DNNDHH/F-D-L/blob/d47939202402696a1ecb63d1cea2c51bca4c0ebc/user.py#L379-L380

**Reply by DNNDHH** [reply]
没什么特别的～本地跑起留有个档…下次看文件就行不用再抓包…懒

**Reply by Zhen-Bo** [reply]
原來如此, 我還以為你跟我一樣都會留一份固定不變的的在本地🤔

**Comment by DNNDHH** [comment]
公钥公开…大家不要搞事情…官方应该短时间内不会换密钥
